### ISSUE TITLE
Usability: OSMand too quick to select 'Select on map'

### ISSUE THREAD

**Issue by ceever** [osmandapp/OsmAnd#16095]
### Description
Getting back to this original issue: https://github.com/osmandapp/OsmAnd/issues/5575

It is good that the automatic routing and zoom out stops when one chooses 'Select on map'. However, in 95% of the time OSMand is far quicker than myself and comes up with the route and zooms out, before one can choose the 'Select on map' option, especially with places nearby. The only way, I can avoid that is by turning off GPS, which is kind of defeating the purpose.

I wouldn't really mind that zooming out, but this feature is the only way to plan routes and hiking trails, by selection a destination first and the starting point later (or vice versa). Especially since now planned routes can be exported/saved in OSMand as gpx route for later, which is a great feature (!) and saves a lot of battery time, it is even more frustrating, that one has to go though this erratic behaviour of the automatic navigation routing.

I guess, especially hikers and mountaineers will use this feature of the gpx export and are plaqued by the automatic zoom out. 

@sonora : Maybe another time, where you can confirm that this is an issue?  

### How to reproduce?
Open OSMand.
Select a nearby POI.
Selection the navigation button.
Try to select "My position" >> 'Select on map' before OSMand comes up with a route. 

### Your Environment
OsmAnd Version: 4.2.7
Android/iOS version: Android 9
Device model: Samsung S8

**Comment by sonora** [comment]
I'm a bit confusef here: When you tap a POI and then select,"Navigation",  then **by far** the most common use case is that you want to go there from where you currently are.

In the few use cases where you actually want to calculate the route from some other location, what's the harm
 - in tapping the blue Navigation button again, then on the route config dialog (which will pop up) tap on the startinng point line, then pick "Select on map"?
 - or simply tap the map screen's navigation button first, then afterwards on the route config dialog tap the starting point first, then the target point last.

Keep in mind that in most use cases people need directions from where they are. Calculating routes from other starting points is a far less frequent use case.

**Comment by ceever** [comment]
Hi @sonora 

Thanks for the quick reply.

The most common use case for car navigation certainly is to actually go there. However, for hiking I beg the differ and I though that was what you actually stated last time, namely that it is more common for hiking to plan a route ahead to know how far it is, how long it takes, or just to export it as a gpx. Right?

The point is that OSMand is way too fast with planning a route when choosing a POI and tapping the navigation button. Then it moves around the map and zooms to cover the route, thus loosing focus on the original map where the user had the relevant POIs.

But you are right, the second option actually is a good way to solve that issue.

Thanks

**Comment by sonora** [comment]
Yes, I fully concur with what you say.

Maybe I am just so used to doing things like I describe that I am not overly sensitive to any difficulty users may have to discover this or a possible improvement, so if you have an idea how things could be fine-tuned, please do suggest!  :wink:

**Comment by ceever** [comment]
You are right, it would just be more useful to maybe wait 3 seconds before OSMand displays the intended route (and zooms out to do that). I think that is enough time for a user to click the "Select on map" option.

So, re-opening the issue with the proposal of a 3 seconds delay before anything is done by OSMand automatically after one selected a POI and hit the Navigation icon.

(For the common user there would be no difference noticeable, because most navigations take longer than 3 seconds anyway to come up with the route, especially car navigation.)

PS: Btw. the routing can already start, but just should not do anything (auto zooming) before not 3 seconds have passed.

**Comment by sonora** [comment]
But the issue was actually fixed, see documentation at #5575? Now once you tap "Select on map", _the map will not re-render or re-position automatically_, that works fine for me.

The only inconvenience case left could be the route finishes calculating so fast that you could not tap "Select on map" beforehand. But then the route distance would be rather short and the corresponding map shift on the screen small anyway, so it's acceptable?

In my opinion it is a bad idea to penalize **every navigation** by 3 seconds,, just because there could be a small fraction of them falling into this residual inconvenience case?

**Comment by ceever** [comment]
Kind of fixed, since I forgot to mention that I am human and do not have superman-like fingers to hit the "Select on map" option in time. ;-)

As I wrote, I am not saying, let's delay everything for 3 seconds. I am just saying to have a buffer of 3 seconds where nothing will happen in the GUI. The routing can already start and even finish within 2 seconds, it should just not be displayed right away.

Unfortunately the map shift is always too extreme, because in most cases my current position is not the point which I would choose on the map through "Select on map".

But you are right, it could be weird if people are used to immediate routing feedback and suddenly it takes 3 seconds. But this is all speculation anyway because we do not know which cases are the more relevant ones.

Maybe an alternative would be to have a option to limit automatic map adjustment in the GUI when not in navigating mode already. Google Maps is similarly annoying sometimes, they also often zoom and relocate to places unintuitively. I guess they think it helps the user, but I feel never did anyone ever investigate such an assumption and developers are just implementing fancy automatic behaviour because it looks nice and feels nice. Yes, fancy things are nice, but are they really convenient?
